tools/disass/radare2/x86.py

```diff
@@ -0,0 +1,167 @@
+      "aac" : "analyze function calls",
+      "aad" : "analyze data references to code",
+      "aae" : "analyze references with ESIL",
+      "aap" : "analyze function preludes"
+    }
+    for x,y in prep_analyses.iteritems():
+        log.info("running command for: {}".format(y))
+        r2.cmd(x)
+    
+    fn_list = r2.cmdj("aflj")
+
+    mod = CFG_pb2.Module()
+    for fne in fn_list:
+        name = fne["name"]
+        new_block = True
+
+        # Having each function be a named block. Is enough?
+        named_block = CFG_pb2.NamedBlock()
+        named_block.name = name
+        named_block.address = fne["offset"]
+
+        # XXX: I am a bit confused on the meaning of the 1 and 0 for
+        # visibility. I am setting the imported symbols to visibility 1.
+        if name.startswith("sym.imp.") == True:
+            named_block.visibility = 1
+        else:
+            named_block.visibility = 0
+        mod.named_blocks.extend([named_block])
+
+        # Analyze the current function
+        r2.cmd("af@{0}".format(name))
+        # Disassemble function and return in json, which 
+        # will look like:
+        # {
+        #   "name":"sym.main",
+        #   "size":261,
+        #   "addr":4195709,
+        #   "ops": [
+        #      {
+        #        "offset":4195709,
+        #        "esil":"rbp,8,rsp,-=,rsp,=[8]",
+        #        "refptr":false,
+        #        "fcn_addr":4195709,
+        #        "fcn_last":4195969,
+        #        "size":1,
+        #        "opcode":"push rbp",
+        #        "bytes":"55",
+        #        "family":"cpu",
+        #        "type":"upush",
+        #        "type_num":12,
+        #        "type2_num":0,
+        #        "flags":["main","sym.main"],
+        #        "xrefs":[{"addr":4195501,"type":"DATA"}]
+        #      }, ... ]
+        # }
+        dis = r2.cmdj("pdfj@{0}".format(name))
+        if dis is None:
+            log.warning("Unable to disassemble {0}".format(name))
+            continue
+
+        for op in dis["ops"]:
+            if new_block == True:
+                if "bytes" not in op.keys():
+                    continue
+                blk = CFG_pb2.Block()
+                blk.address = op["offset"]
+                new_block = False
+           
+            if "bytes" not in op.keys():
+                if new_block == False:
+                    mod.blocks.extend([blk])
+                    new_block = True
+                    blk = None
+                continue
+
+            ii = CFG_pb2.Instr()
+
+            # I was using python3 before so had bytes.fromhex() API call,
+            # but I believe this is the correct juju for Python2
+            btmp = [op["bytes"][i:i+2] for i in range(0, len(op["bytes"]), 2)]
+            ii.bytes = ''.join([chr(int(x, 16)) for x in btmp])
+            ii.address = op["offset"]    
+            oc = op["opcode"]
+
+            #
+            # Unsure if this is the behavior that is desired
```

Comment:
`ud2` is a valid block terminator, and has well-defined semantics in Remill. A block ending in a `ud2` should be included in the CFG.